ui/pages/confirmations/hooks/useTypedSignSignatureInfo.ts

```diff
@@ -0,0 +1,46 @@
+import { useMemo } from 'react';
+
+import {
+  isOrderSignatureRequest,
+  isPermitSignatureRequest,
+  isSignatureTransactionType,
+} from '../utils';
+import { SignatureRequestType } from '../types/confirm';
+import { parseTypedDataMessage } from '../../../../shared/modules/transaction.utils';
+import { TokenStandard } from '../../../../shared/constants/transaction';
+import { MESSAGE_TYPE } from '../../../../shared/constants/app';
+import { TypedSignSignaturePrimaryTypes } from '../constants';
+
+export const useTypedSignSignatureInfo = (
+  confirmation: SignatureRequestType,
+) => {
+  if (!confirmation) {
+    return {};
```

Comment:
Minor, would it be more explicit to return `undefined` here so we can insist primary type and standard are always returned?

Reply:
I do not get this point exactly, I return empty object here for ease of destructuring the result.